pkg/docker/docker.go

```diff
@@ -18,41 +18,91 @@ import (
 	docker "github.com/docker/docker/client"
 	"github.com/docker/docker/pkg/archive"
 	"github.com/kyma-project/cli/internal/minikube"
+	"github.com/kyma-project/cli/pkg/step"
 )
 
 const (
 	defaultRegistry = "index.docker.io"
 )
 
-// DockerErrorMessage is used to parse error messages coming from Docker
-type DockerErrorMessage struct {
+type dockerClient struct {
+	*docker.Client
+	// DockerClientService
+}
+
+type kymaDockerClient struct {
+	kymaDocker ClientService
+}
+
+//go:generate mockery --name ClientService
+type ClientService interface {
+	ArchiveDirectory(srcPath string, options *archive.TarOptions) (io.ReadCloser, error)
+	NegotiateAPIVersion(ctx context.Context)
+	ImagePush(ctx context.Context, image string, options types.ImagePushOptions) (io.ReadCloser, error)
+	ImageBuild(ctx context.Context, buildContext io.Reader, options types.ImageBuildOptions) (types.ImageBuildResponse, error)
+}
+
+type KymaDockerService interface {
+	PushKymaInstaller(image string, currentStep step.Step) error
+	BuildKymaInstaller(localSrcPath, imageName string) error
+}
+
+// ErrorMessage is used to parse error messages coming from Docker
+type ErrorMessage struct {
 	Error string
 }
 
-func (i *Installation) buildKymaInstaller(imageName string) error {
-	var dc *docker.Client
-	var err error
-	if i.Options.IsLocal {
-		dc, err = minikube.DockerClient(i.Options.Verbose, i.Options.LocalCluster.Profile, i.Options.Timeout)
-	} else {
-		dc, err = docker.NewClientWithOpts(docker.FromEnv)
+//NewDockerService creates docker client using docker environment of the OS
+func NewDockerService() (ClientService, error) {
+	dClient, err := docker.NewClientWithOpts(docker.FromEnv)
+	if err != nil {
+		return nil, err
 	}
+
+	return &dockerClient{
+		dClient,
+	}, nil
+}
+
+//NewDockerMinikubeService creates docker client for minikube docker-env
+func NewDockerMinikubeService(verbosity bool, profile string, timeout time.Duration) (ClientService, error) {
+	dClient, err := minikube.DockerClient(verbosity, profile, timeout)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	reader, err := archive.TarWithOptions(i.Options.LocalSrcPath, &archive.TarOptions{})
+	return &dockerClient{
+		dClient,
+	}, nil
+}
+
+func NewKymDockerClientService(isLocal bool, verbosity bool, profile string, timeout time.Duration) (KymaDockerService, error) {
```

Comment:
There is a typo here:
```suggestion
func NewKymaDockerClientService(isLocal bool, verbosity bool, profile string, timeout time.Duration) (KymaDockerService, error) {
```

Reply:
done.